### ISSUE TITLE
please add support for tinipic

### ISSUE THREAD

**Issue by treaki** [rg3/youtube-dl#4754]
```
treaki@hostname:/tmp$ youtube-dl "http://de.tinypic.com/player.php?v=dy90yh&s=8"
[generic] player: Requesting header
WARNING: Falling back on generic information extractor.
[generic] player: Downloading webpage
[generic] player: Extracting information
ERROR: Unsupported URL: http://de.tinypic.com/player.php?v=dy90yh&s=8
treaki@hostname:/tmp$ youtube-dl --verbose http://de.tinypic.com/r/dy90yh/8
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['--verbose', 'http://de.tinypic.com/r/dy90yh/8']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.01.16
[debug] Python version 2.7.8 - Linux-3.16-3-amd64-x86_64-with-debian-jessie-sid
[debug] exe versions: avconv 2.4.3, avprobe 2.4.3, ffmpeg 2.4.3, ffprobe 2.4.3, rtmpdump 2.4
[debug] Proxy map: {}
[generic] 8: Requesting header
[redirect] Following redirect to http://de.tinypic.com/player.php?v=dy90yh&s=8
[generic] player: Requesting header
WARNING: Falling back on generic information extractor.
[generic] player: Downloading webpage
[generic] player: Extracting information
ERROR: Unsupported URL: http://de.tinypic.com/player.php?v=dy90yh&s=8
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/generic.py", line 670, in _real_extract
    doc = parse_xml(webpage)
  File "/usr/local/bin/youtube-dl/youtube_dl/utils.py", line 1504, in parse_xml
    tree = xml.etree.ElementTree.XML(s.encode('utf-8'), **kwargs)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1300, in XML
    parser.feed(text)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1642, in feed
    self._raiseerror(v)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1506, in _raiseerror
    raise err
ParseError: not well-formed (invalid token): line 4, column 452
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 600, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 260, in extract
    return self._real_extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/generic.py", line 1075, in _real_extract
    raise UnsupportedError(url)
UnsupportedError: Unsupported URL: http://de.tinypic.com/player.php?v=dy90yh&s=8

treaki@hostname:/tmp$  
```

**Comment by dstftw** [comment]
Will be fixed in the next version. Thanks for the report.

**Comment by treaki** [comment]
```
treaki@hostname:/tmp/tpv$ wget "http://de.tinypic.com/r/dy90yh/8"
--2015-01-20 21:22:47--  http://de.tinypic.com/r/dy90yh/8
Resolving de.tinypic.com (de.tinypic.com)... 209.17.68.209
Connecting to de.tinypic.com (de.tinypic.com)|209.17.68.209|:80... connected.
HTTP request sent, awaiting response... 302 Found
Location: /player.php?v=dy90yh&s=8 [following]
--2015-01-20 21:22:48--  http://de.tinypic.com/player.php?v=dy90yh&s=8
Reusing existing connection to de.tinypic.com:80.
HTTP request sent, awaiting response... 200 OK
Length: 38029 (37K) [text/html]
Saving to: ‘8’

8                               100%[=======================================================>]  37.14K  55.2KB/s   in 0.7s   

2015-01-20 21:22:49 (55.2 KB/s) - ‘8’ saved [38029/38029]

treaki@hostname:/tmp/tpv$ less 8 
treaki@hostname:/tmp/tpv$ less 8 
treaki@hostname:/tmp/tpv$ grep "fo.addVariable("file"," 8
treaki@hostname:/tmp/tpv$ grep 'fo.addVariable("file",' 8
                    fo.addVariable("file", "dy90yh");
treaki@hostname:/tmp/tpv$ grep 'fo.addVariable(' 8 
                fo.addVariable("ipt", "%3Cembed+width%3D%22440%22+height%3D%22420%22+type%3D%22application%2Fx-shockwave-flash%22+src%3D%22http%3A%2F%2Fv8.tinypic.com%2Fplayer.swf%3Ffile%3Ddy90yh%26s%3D8%22%3E%3Cbr%3E%3Cfont+size%3D%221%22%3E%3Ca+href%3D%22http%3A%2F%2Fde.tinypic.com%2Fplayer.php%3Fv%3Ddy90yh%26s%3D8%22%3EOriginal+Video%3C%2Fa%3E+-+More+videos+at+%3Ca+href%3D%22http%3A%2F%2Ftinypic.com%22%3ETinyPic%3C%2Fa%3E%3C%2Ffont%3E");
                fo.addVariable("trk", "copy_video_html");
                fo.addVariable("ipt", "%5BURL%3Dhttp%3A%2F%2Fde.tinypic.com%2Fr%2Fdy90yh%2F8%5DView+My+Video%5B%2FURL%5D");
                fo.addVariable("trk", "copy_video_img");
                fo.addVariable("ipt", "http%3A%2F%2Fde.tinypic.com%2Fr%2Fdy90yh%2F8");
                fo.addVariable("trk", "copy_video_view");
                    fo.addVariable("file", "dy90yh");
                    fo.addVariable("s", "8");
                    fo.addVariable("ap", "1");
                    fo.addVariable("nt", "0");
                    fo.addVariable("os", "1");
                    fo.addVariable("fms", "1");
treaki@hostname:/tmp/tpv$ wget http://v8.tinypic.com/dy90yh.flv
--2015-01-20 21:25:54--  http://v8.tinypic.com/dy90yh.flv
Resolving v8.tinypic.com (v8.tinypic.com)... 209.17.68.209
Connecting to v8.tinypic.com (v8.tinypic.com)|209.17.68.209|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 425178 (415K) [application/octet-stream]
Saving to: ‘dy90yh.flv’

dy90yh.flv                      100%[=======================================================>] 415.21K  41.6KB/s   in 11s    

2015-01-20 21:26:06 (36.9 KB/s) - ‘dy90yh.flv’ saved [425178/425178]

```

looks like it would be easy to implement, just use following pattern:

http://v%servernumber%.tinypic.com/%filekey%.flv

and replace %servernumber% with that what is written in fo.addVariable("s"
and replace %filekey% ith that what is written in fo.addVariable("file",

**Comment by treaki** [comment]
ok, thanks a loot. i am looking forward to see that in youtube-dl

**Comment by phihag** [comment]
The fix made it into youtube-dl 2015.01.23. See [our FAQ](https://yt-dl.org/update) if you need help updating.

**Comment by treaki** [comment]
thanks a loot, ill try that later the day. youtube-dl gets updated by a cron job on my box :)
